File: Aut_of_risk_determination_BE/src/main/java/aord/model/DangerDetails.java
package aord.model;

import javax.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;
import java.util.Set;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "DangerDetails")
public class DangerDetails {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "probability_id")
    private Probability probability;

    @ManyToOne
    @JoinColumn(name = "effect_id")
    private Effect effect;

    @ManyToOne
    @JoinColumn(name = "frequency_id")
    private Frequency frequency;

    @ManyToOne
    @JoinColumn(name = "danger_name_id")
    private DangerName dangerName;

    @Column
    private Integer code;

    @Column
    private Double value;

    @Column
    @ElementCollection(targetClass = String.class)
    private List<String> descriptions;

}

File: Aut_of_risk_determination_FE/aut_of_risk_determination/src/app/model/DangerDetails.ts
import { Probability } from "./Probability";
import { Effect } from "./Effect";
import { Frequency } from "./Frequency";
import { DangerName } from "./DangerName";

export class DangerDetails {
    constructor(
        public id: number,
        public code?: number,
        public value?: number,
        public descriptions?: string[],
        public probability?: Probability,
        public effect?: Effect,
        public frequency?: Frequency,
        public dangerName?: DangerName
    ) {

    }
}